quickwit-indexing/src/source/kinesis/shard_consumer.rs

```diff
@@ -93,10 +95,39 @@ impl ShardConsumer {
             sink,
         }
     }
+
+    pub fn spawn(self, ctx: &SourceContext) -> ShardConsumerHandle {
+        let shard_id = self.shard_id.clone();
+        let (mailbox, actor_handle) = ctx
+            .spawn_actor(self)
+            .set_kill_switch(ctx.kill_switch().clone())
+            .spawn_async();
+        ShardConsumerHandle {
+            shard_id,
+            mailbox,
+            actor_handle,
+        }
+    }
+
+    async fn send_message(
+        &self,
+        ctx: &ActorContext<Self>,
+        message: ShardConsumerMessage,
+    ) -> anyhow::Result<()> {
+        let _guard = ctx.protect_zone();
+        self.sink.send(message).await?;
+        Ok(())
+    }
+}
+
+pub(super) struct ShardConsumerHandle {
+    pub shard_id: String,
+    pub mailbox: Mailbox<Loop>,
+    pub actor_handle: ActorHandle<ShardConsumer>,
 }
 
 #[derive(Debug)]
-struct Loop;
+pub(super) struct Loop;
 
 impl Actor for ShardConsumer {
     type Message = Loop;
```

Comment:
Sorry the API has changed now. 

impl Handler<Loop> for ShardConsumer {
  //..
}